### ISSUE TITLE
URL encoding is inconsistent between works and authors and indexes

### ISSUE THREAD

**Issue by YairRand** [Sefaria/Sefaria-Project#348]
For works (eg `/Mishneh_Torah,_Transmission_of_the_Oral_Law.1`), the canonical URLs encode spaces as underscores. For indices (eg `/texts/Halakhah/Mishneh Torah`) and authors (eg `/person/Israel Meir Kagan`), spaces are encoded as plain spaces/`%20`, and underscores aren't redirected. This can be confusing on occasion.

**Comment by JonMosenkis** [comment]
I agree. Version Titles have this problem as well. We handle spaces/underscores when we do Ref parsing, but it might be worth considering swapping out spaces for underscores when we parse our urls.

**Comment by YairRand** [comment]
Related: Even within the author pages, all links to works by the author use spaces, requiring redirecting after clicking. (For example, `/person/Ibn%20Ezra` links to `/Ibn%20Ezra%20on%20Genesis`, which then redirects to `/Ibn_Ezra_on_Genesis`.)
